src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx

```diff
@@ -1,23 +1,182 @@
 
 const displayName = 'v5.pages.UserPreferencesPage';
 
 const UserPreferencesPage = () => {
   const { user } = useCanEditProfile();
+  const isMobile = useMobile();
+  const { formatMessage } = useIntl();
+  const [isEmailInputVisible, setIsEmailInputVisible] = useState(false);
+  const { handleClipboardCopy } = useCopyToClipboard(user?.walletAddress || '');
+
+  const { errors, handleSubmit, onSubmit, register, getValues } =
+    useUserPreferencesPage();
 
   if (!user) {
     return null;
   }
 
+  const emailValue = getValues('email');
+
   return (
     <Spinner loadingText={{ id: 'loading.userPreferencesPage' }}>
       <TwoColumns aside={<Navigation pageName="profile" />}>
-        Preferences
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div className="flex flex-col gap-6">
+            <h4 className="heading-4">
+              {formatMessage({ id: 'userPreferencesPage.accountPreferences' })}
+            </h4>
+            <div className={styles.row}>
+              <LeftColumn
+                fieldTitle={{ id: 'field.email' }}
+                fieldDescription={{ id: 'description.email' }}
+              />
+              <div className="w-full flex flex-col">
+                {emailValue && !isEmailInputVisible && (
+                  <div className="flex items-center justify-end">
+                    <span className="text-md mr-6">{emailValue}</span>
+                    <Button
+                      mode="primarySolid"
+                      text={{ id: 'button.updateEmail' }}
+                      onClick={() => setIsEmailInputVisible(true)}
+                    />
+                  </div>
+                )}
+                {!emailValue || isEmailInputVisible ? (
+                  <>
+                    <Input
+                      name="email"
+                      register={register}
+                      isError={!!errors.email?.message}
+                      customErrorMessage={errors.email?.message}
+                    />
+                    <div className="ml-auto mt-2">
+                      <Button
+                        mode="primarySolid"
+                        text={{ id: 'button.saveEmail' }}
+                        type="submit"
+                        onClick={() => setIsEmailInputVisible(false)}
+                      />
+                    </div>
+                  </>
+                ) : undefined}
+              </div>
+            </div>
+            <span className="divider" />
+            <div className={styles.row}>
+              <div className="flex md:flex-row flex-col items-center justify-between p-3 bg-gray-50 rounded-lg w-full">
+                <div className="flex items-center mb-3 md:mb-0">
+                  <Icon name="cardholder" appearance={{ size: 'small' }} />
+                  <span className="text-md ml-2 truncate block w-full">
+                    {isMobile
+                      ? multiLineTextEllipsis(user.walletAddress, 20)
```

Comment:
maybe better will be to pass the second parameter as a optional props added to UserPreferencesPage component and with default value 20?